test/timezone/lookup/test_geonames.rb

```diff
@@ -35,6 +49,19 @@ def test_lookup
     assert_equal 'Australia/Adelaide', mine.lookup(*coordinates)
   end
 
+  def test_lookup_with_etc
+    %w(arctic atlantic).each do |point|
```

Comment:
If you convert the data into a more structured hash this can be simplified.

```
def etc_data
  {
    arctic: { coordinates: [89, 40], name: 'Etc/GMT+3' },
    atlantic: {... }
  }
end

def test_lookup_with_etec
  etc_data.each do |key, data|
    ...
    assert_equal data[:name], mine.lookup(*data[:coordinates])
```